schema.graphql

```diff
@@ -229,57 +263,113 @@ input AddOpticalImageInput {
-  datasetByName(name: String!): Dataset
+  # If JWT is missing, only public information is returned;
+  # otherwise, user access rights are taken into account;
+  # For consistency, every query has 'jwt' parameter.
+
+  dataset(jwt: String, id: String!): Dataset
+  user(jwt: String, id: String!): UserProfile
+  group(jwt: String, id: String!): Group
+  project(jwt: String, id: String!): Project
+
+  datasetByName(jwt: String, name: String!): Dataset
 
   allDatasets(orderBy: DatasetOrderBy = ORDER_BY_DATE,
               sortingOrder: SortingOrder = DESCENDING,
               filter: DatasetFilter = {}, simpleQuery: String,
-              offset: Int = 0, limit: Int = 10): [Dataset!]!
+              offset: Int = 0, limit: Int = 10, jwt: String): [Dataset!]!
 
   allAnnotations(orderBy: AnnotationOrderBy = ORDER_BY_MSM,
                  sortingOrder: SortingOrder = DESCENDING,
                  offset: Int = 0, limit: Int = 10,
                  filter: AnnotationFilter = {},
                  datasetFilter: DatasetFilter = {},
-                 simpleQuery: String): [Annotation]
+                 simpleQuery: String, jwt: String): [Annotation]
 
-  countDatasets(filter: DatasetFilter = {}, simpleQuery: String): Int!
+  countDatasets(filter: DatasetFilter = {}, simpleQuery: String, jwt: String): Int!
 
-  countDatasetsPerGroup(query: DatasetCountPerGroupInput!): DatasetCountPerGroup!
+  countDatasetsPerGroup(query: DatasetCountPerGroupInput!, jwt: String): DatasetCountPerGroup!
 
   countAnnotations(filter: AnnotationFilter = {},
                    datasetFilter: DatasetFilter = {},
-                   simpleQuery: String): Int!
+                   simpleQuery: String, jwt: String): Int!
 
-  annotation(id: String!): Annotation
+  annotation(id: String!, jwt: String): Annotation
 
-  metadataSuggestions(field: String!, query: String!, limit: Int = 10): [String!]!
+  metadataSuggestions(field: String!, query: String!, limit: Int = 10, jwt: String): [String!]!
 
-  peopleSuggestions(role: PersonRole!, query: String!): [Person!]!
+  peopleSuggestions(role: PersonRole!, query: String!, jwt: String): [Person!]!
 
-  molecularDatabases: [MolecularDatabase!]
+  molecularDatabases(jwt: String): [MolecularDatabase!]
 
-  opticalImageUrl(datasetId: String!, zoom: Float = 1): String
+  opticalImageUrl(datasetId: String!, zoom: Float = 1, jwt: String): String
 
-  rawOpticalImage(datasetId: String!): RawOpticalImage
+  rawOpticalImage(datasetId: String!, jwt: String): RawOpticalImage
+}
+
+input UserProfileInput {
+  id: String  # can differ from the id stored in the JWT, as group/project managers should be able to edit members' profiles
+  name: String!
+  email: String!
+}
+
+input DatasetInput {
+  id: String!
+  name: String!
+  path: String!
+  private: Boolean = true
+  sync: Boolean = true
+  priority: Int = 0
+  deleteFirst: Boolean = false
+  metadataJson: String  # absent => use currently stored metadata
+}
+
+enum Role {
```

Comment:
@lomereiter do we need PersonRole type now?

Reply:
I think these can be removed or merged somehow.
@andy-d-palmer do we want to have first name and surname as two fields, or just fullName would do? Having them separate doesn't seem to be terribly useful.

Reply:
@lomereiter merging names into one fullName field seems fine to me